### PR TITLE
Batch recording. Closes #6

### LINKED ISSUE #6: Add support for recording multiple network calls in one test run
Since `Cassette` saves an array of `Interaction`s, instead of just one, it would be great to add support for recording and replaying multiple interactions in one test case.

Possible changes for recording:
- add a property on `Session` called `recordMultiple`, which would allow for not stopping after the first `Interaction` has been recorded
- add a method on `Session` called `finishRecording()`, which persists all `Interaction`s from memory to disk and asserts to present the file path, like it does now. `finishRecording()` would only have to be called if `recordMultiple` is set to `true`
- maybe even call `finishRecording()` from `deinit` of `Session`, so that user doesn't have to call it manually

Possible changes for replaying:
- keep counter of which `Interaction` should be played next, automatically increment after each replay
- assert on `deinit` that exactly the number of prerecorded `Interaction`s has been called (fail the test if more or less network calls were made)

???

Profit.

### PR DESCRIPTION
**Pull request by soffes**
Support for recording multiple interactions. This also includes #29 since the recorded cassettes had plain text in them.

### CODE REVIEW

**Comment by soffes** [comment]
I've tested this throughly both in DVR and in an application using DVR. Everything works as expected!

Here's an example. In this client, it's one request to get a token and one request to show the user to build a full Account (token + user). This passes as expected with this change :grinning: 

``` swift
func testSignIn() {
    let expectation = expectationWithDescription("Networking")

    let dvr = Session(cassetteName: "sign-in")
    dvr.beginRecording()

    let client = AuthorizationClient(session: dvr)

    client.signIn(username: "soffes", password: "REDACTED_PASSWORD") {
        switch $0 {
        case .Success(let account):
            XCTAssertEqual("REDACTED_TOKEN", account.accessToken)
            XCTAssertEqual("soffes", account.user.username)
        case .Failure(let message):
            XCTFail(message)
        }

        dvr.endRecording() {
            expectation.fulfill()
        }
    }

    waitForExpectationsWithTimeout(1, handler: nil)
}
```

File: DVR/Session.swift [hunk 1]
```diff
@@ -25,34 +32,155 @@ public class Session: NSURLSession {
...
+			endRecording()
+		}
+	}
+
+	private func persist(interactions: [Interaction]) {
+		// Create directory
+		let outputDirectory = (self.outputDirectory as NSString).stringByExpandingTildeInPath
+		let fileManager = NSFileManager.defaultManager()
+		if !fileManager.fileExistsAtPath(outputDirectory) {
+			try! fileManager.createDirectoryAtPath(outputDirectory, withIntermediateDirectories: true, attributes: nil)
```
Comment (ayanonagon) [inline DVR/Session.swift]: Can we catch this? :D

Comment (soffes) [inline DVR/Session.swift]: It was already like that, but I'll add a catch.

**Comment by soffes** [comment]
@ayanonagon ready to go!

**Pull request merged**